### PR TITLE
stkAAVE Rewards Update

### PR DESCRIPTION
**Pull request by efecarranza**
# Changelog

Extend stkAAVE rewards.

### CODE REVIEW

File: src/20241106_AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards/AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards_20241106.sol [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  uint128 public constant AAVE_EMISSION_PER_SECOND_STK_AAVE = uint128(360e18) / 1 days; // 360 AAVE per day
+
+  function execute() external {
+    IStakeToken.AssetConfigInput[] memory aaveConfigs = new IStakeToken.AssetConfigInput[](1);
+    aaveConfigs[0] = IStakeToken.AssetConfigInput({
+      emissionPerSecond: AAVE_EMISSION_PER_SECOND_STK_AAVE,
+      totalStaked: 0, // it's overwritten internally
+      underlyingAsset: AaveSafetyModule.STK_AAVE
+    });
+    IStakeToken(AaveSafetyModule.STK_AAVE).configureAssets(aaveConfigs);
```
[the remark this answers is not in the document]
Comment (brotherlymite) [reply src/20241106_AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards/AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards_20241106.sol:29]: Reference: https://explorer.sim.io/eth/21149398/0x4da27a545c0c5b758a6ba100e3a049001de870f5/assets.0x4da27a545c0c5b758a6ba100e3a049001de870f5#map

File: src/20241106_AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards/SafetyModuleStkAAVEReEnableRewards.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+## Specification
+
+This proposal will implement the following changes to the AAVE emissions across the three SM categories:
+
+| Asset   | Current AAVE/Day | Proposed AAVE/Day |
+| ------- | ---------------- | ----------------- |
+| stkAAVE | 360              | 360               |
+
+The distribution end for 180 days from Execution date.
```
Comment (brotherlymite) [inline src/20241106_AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards/SafetyModuleStkAAVEReEnableRewards.md:26]: @efecarranza The distributionEnd cannot be changed in stkAave, it is set as an immutable

Comment (efecarranza) [reply src/20241106_AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards/SafetyModuleStkAAVEReEnableRewards.md:26]: removed

File: src/20241106_AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards/AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards_20241106.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (efecarranza) [reply src/20241106_AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards/AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards_20241106.sol:29]: @brotherlymite i had read it's done to update the internal state:

https://github.com/bgd-labs/aave-proposals-v3/pull/506/files#diff-d188e9df06382a22d0a34b45870436aff2b67b1880843437016652451aabb08aR24

[the remark this answers is not in the document]
Comment (brotherlymite) [reply src/20241106_AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards/AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards_20241106.sol:29]: The distributionEnd was never halted, the users are accruing rewards fine just due to allowance not able to claim them, so updating the state should not be required in this case

File: src/20241106_AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards/SafetyModuleStkAAVEReEnableRewards.md (hunk 2 above)
Comment (brotherlymite) [reply src/20241106_AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards/SafetyModuleStkAAVEReEnableRewards.md:26]: I think the changes were not pushed 😅

File: src/20241106_AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards/AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards_20241106.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (ianflexa) [reply src/20241106_AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards/AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards_20241106.sol:29]: As @brotherlymite mentioned, it just needs to update the allowance. Users will update the state when interacting with it

[the remark this answers is not in the document]
Comment (efecarranza) [reply src/20241106_AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards/AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards_20241106.sol:29]: thank you!

File: src/20241106_AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards/AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards_20241106.sol [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import {AaveSafetyModule} from 'aave-address-book/AaveSafetyModule.sol';
+import {IProposalGenericExecutor} from 'aave-helpers/src/interfaces/IProposalGenericExecutor.sol';
+
+import {IStakeToken} from 'aave-address-book/common/IStakeToken.sol';
+
+/**
+ * @title Safety Module stkAAVE - Re-enable Rewards
+ * @author @karpatkey_TokenLogic
+ * - Snapshot: Direct-to-AIP
+ * - Discussion: https://governance.aave.com/t/arfc-amend-safety-module-emissions/16640/12
```
Comment (ianflexa) [inline src/20241106_AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards/AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards_20241106.sol:16]:
```suggestion
 * - Discussion: https://governance.aave.com/t/arfc-amend-safety-module-emissions/16640/13
```

File: src/20241106_AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards/SafetyModuleStkAAVEReEnableRewards.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
+---
+title: "Safety Module stkAAVE - Re-enable Rewards"
+author: "@karpatkey_TokenLogic"
+discussions: ""
```
Comment (ianflexa) [inline src/20241106_AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards/SafetyModuleStkAAVEReEnableRewards.md:4]:
```suggestion
discussions: "https://governance.aave.com/t/arfc-amend-safety-module-emissions/16640/13"
```

File: src/20241106_AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards/SafetyModuleStkAAVEReEnableRewards.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+| Asset   | Current AAVE/Day | Proposed AAVE/Day |
+| ------- | ---------------- | ----------------- |
+| stkAAVE | 360              | 360               |
+
+## References
+
+- Implementation: [AaveV3Ethereum](https://github.com/bgd-labs/aave-proposals-v3/blob/main/src/20241106_AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards/AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards_20241106.sol)
+- Tests: [AaveV3Ethereum](https://github.com/bgd-labs/aave-proposals-v3/blob/main/src/20241106_AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards/AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards_20241106.t.sol)
+- [Snapshot](Direct-to-AIP)
+- [Discussion](https://governance.aave.com/t/arfc-amend-safety-module-emissions/16640/12)
```
Comment (ianflexa) [inline src/20241106_AaveV3Ethereum_SafetyModuleStkAAVEReEnableRewards/SafetyModuleStkAAVEReEnableRewards.md:31]:
```suggestion
- [Discussion](https://governance.aave.com/t/arfc-amend-safety-module-emissions/16640/13)
```

**Review by ianflexa — APPROVED** [review]

**Review by brotherlymite — APPROVED** [review]

**Pull request merged**